### PR TITLE
Add some new REST methods, test them

### PR DESCRIPTION
**Pull request by cory-signalfx**
# Summary

Adds new methods to the REST client:

* `get_detector(id)`
* `get_detector_incidents(id)`
* `get_incidents`
* `clear_incident`

# Motivation

I was working on some stuff and using the Python library. Turns out it was missing some stuff!

# Notes

I added unit tests using HTTMock. I did this cuz it's what I found in doing some searches. I've attempted to add the requirements, etc but if there are better ways I'm all for 'em. The tests themselves use JSON files for source, to save me typing.

cc @AlmightyOatmeal

### CODE REVIEW

File: signalfx/rest.py [hunk 1]
```diff
@@ -427,3 +435,47 @@ def delete_detector(self, detector_id, **kwargs):
...
+
+    # functionality related to incidents
+    def get_incident(self, id, **kwargs):
+        """"Retrieve a (v2) incident by id.
+        """
+        resp = self._get_object_by_name(self._INCIDENT_ENDPOINT_SUFFIX, id,
+                                        **kwargs)
+        return resp
+
+    def get_incidents(self, detector_id=None, offset=0, limit=None, include_resolved=False, **kwargs):
```
Comment (rmfitzpatrick) [inline signalfx/rest.py]: Where are `detector_id` and `include_resolved` used?

File: tests/unittests_test.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+def get_detector_by_id(url, request):
+    if '/v2/detector/abc123' not in url.path:
+        return {
+            'status_code': 400,
+            'content': 'Incorrect URL: ' + url.path
+        }
+
+    return {
+        'status_code': 200,
+        'content': open('tests/fixtures/get_detector_abc123.json').read(),
```
Comment (rmfitzpatrick) [inline tests/unittests_test.py]: Looks like each file is read twice.  Any reason not to consolidate?

**Comment by cory-signalfx** [comment]
Hey @rmfitzpatrick, this is ready for another look!

File: tests/unittests_test.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        if spec is None:
+            raise Exception("Unknown mock")
+
+        if spec['path'] != url.path or request.method != spec['method']:
+            return {
+                'content': 'Unknown URL' + url.path,
+                'status_code': 400,
+                }
+
+        if spec['params'] is not None and spec['params'] != url.query:
```
Comment (rmfitzpatrick) [inline tests/unittests_test.py]: `spec.get('params')`?  Only `GET_INCIDENTS` has them set.

File: tests/unittests_test.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            with signalfx.SignalFx().rest('authkey') as sfx:
+                orig_incidents = json.loads(responses_map[name]['content'])
+                incidents = sfx.get_incidents(include_resolved=True)
+                self.assertEqual(len(orig_incidents), len(incidents))
+
+    def test_get_incident(self):
+        name = 'GET_INCIDENT'
+        with HTTMock(mock_maker(name)):
+            with signalfx.SignalFx().rest('authkey') as sfx:
+                print(responses_map[path])
```
Comment (rmfitzpatrick) [inline tests/unittests_test.py]: old debug statement?

File: tests/unittests_test.py (hunk 3 above)
Comment (cory-signalfx) [reply tests/unittests_test.py]: Changed, to `get`, but not sure I follow the ask. Yes, only that one case has them set hence the `None` check.

Comment (rmfitzpatrick) [reply tests/unittests_test.py]: Right, just referring to the resulting key error with the direct access.

File: tests/unittests_test.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2017 SignalFx, Inc. All rights reserved.
+
+from httmock import all_requests, urlmatch, HTTMock
+# from urllib.parse import urlparse
+import json
+import unittest
+import urlparse
```
Comment (rmfitzpatrick) [inline tests/unittests_test.py]: You can get this via `six.moves.urllib`

File: tests/unittests_test.py [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+        if spec.get('path') != url.path or request.method != spec.get('method'):
+            return {
+                'content': 'Unknown URL' + url.path,
+                'status_code': 400,
+                }
+
+        if spec.get('params') is not None:
+            incoming_params = urlparse.parse_qs(url.query)
+
+            for k, v in spec.get('params').iteritems():
```
Comment (rmfitzpatrick) [inline tests/unittests_test.py]: iteritems is python 2 only (sry if duplicate, gh is loading oddly for me)

**Comment by cory-signalfx** [comment]
Fixups force pushed! (I wasn't sure of the philosophies at play here!)

**Pull request merged**